File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/HistoryApi.swift
import Foundation

/// Facade for activity history operations.
public struct HistoryApi {
    private let repo: ActivityRepository

    init(repo: ActivityRepository) {
        self.repo = repo
    }

    public func log(_ record: ActivityRecord) async throws {
        try await repo.addActivity(record)
    }

    public func list() async -> [ActivityRecord] {
        await repo.listActivities()
    }

    public func listForDid(_ did: String) async -> [ActivityRecord] {
        await repo.listActivitiesForDid(did)
    }

    public func clearAll() async throws {
        try await repo.clearAll()
    }
}

File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/HistoryApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.history.ActivityRepository
import my.ssdid.sdk.domain.model.ActivityRecord

class HistoryApi internal constructor(private val repo: ActivityRepository) {
    suspend fun log(record: ActivityRecord) = repo.addActivity(record)
    suspend fun list(): List<ActivityRecord> = repo.listActivities()
    suspend fun listForDid(did: String): List<ActivityRecord> = repo.listActivitiesForDid(did)
    suspend fun clearAll() = repo.clearAll()
}
